extensions/metrics/src/interceptors/metrics.interceptor.ts

```diff
@@ -74,22 +86,41 @@ export class MetricsInterceptor implements Provider<Interceptor> {
     next: () => ValueOrPromise<T>,
   ) {
     MetricsInterceptor.setup();
-    const endGauge = MetricsInterceptor.gauge.startTimer({
-      targetName: invocationCtx.targetName,
-    });
-    const endHistogram = MetricsInterceptor.histogram.startTimer({
-      targetName: invocationCtx.targetName,
-    });
-    const endSummary = MetricsInterceptor.summary.startTimer({
+    const {request, response} = invocationCtx.parent as RequestContext;
+    const labelValues: LabelValues<LabelNames> = {
       targetName: invocationCtx.targetName,
-    });
+      method: request.method,
+      path: request.path,
+    };
+    const endGauge = MetricsInterceptor.gauge.startTimer();
+    const endHistogram = MetricsInterceptor.histogram.startTimer();
+    const endSummary = MetricsInterceptor.summary.startTimer();
     try {
-      MetricsInterceptor.counter.inc();
-      return await next();
+      const result = await next();
+      labelValues.statusCode = getStatusCodeFromResponse(response, result);
+      return result;
+    } catch (err) {
+      labelValues.statusCode = getStatusCodeFromError(err);
+      throw err;
     } finally {
-      endGauge();
-      endHistogram();
-      endSummary();
+      MetricsInterceptor.counter.inc(labelValues);
+      endGauge(labelValues);
+      endHistogram(labelValues);
+      endSummary(labelValues);
     }
   }
 }
+
+function getStatusCodeFromResponse(response: Response, result?: unknown) {
+  // interceptors are invoked before result is written to response,
+  // the status code for 200 responses without a result should be 204
+  const noContent = response.statusCode === 200 && result === undefined && 204;
+  return noContent || response.statusCode;
+}
+
+function getStatusCodeFromError(err: HttpErrors.HttpError) {
+  // interceptors are invoked before error is written to response,
+  // it is required to retrieve the status code the error
+  const notFound = err.code === 'ENTITY_NOT_FOUND' && 404;
+  return err.statusCode || err.status || notFound || 500;
```

Comment:
`return err.statusCode ?? err.status ?? notFound ?? 500;`?

Reply:
same here `notFound` is of type `false | 404`, I could change it to `return err.statusCode ?? err.status ?? (notFound || 500);` does that make sense?

Reply:
same here function implementation could be changed to 

```ts
function getStatusCodeFromError(err: HttpErrors.HttpError) {
  const notFound = err.code === 'ENTITY_NOT_FOUND';
  return err.statusCode ?? err.status ?? (notFound ? 404 : 500);
}
```